pkg/manager/member/tidb_member_manager.go

```diff
@@ -288,13 +289,17 @@ func (tmm *tidbMemberManager) syncTiDBServerCerts(tc *v1alpha1.TidbCluster) erro
 		fmt.Sprintf("%s.%s.svc", svcName, ns),
 		"localhost",
 	}
-	hostList = append(hostList, tc.Spec.TiDB.ExtraSANDomainList...)
+	if tc.Spec.TiDB.TLSClient.AutoGenerated != nil {
+		hostList = append(hostList, tc.Spec.TiDB.TLSClient.AutoGenerated.ExtraSANDomainList...)
+	}
 
 	ipList := []string{
 		"127.0.0.1", "::1",
 		svc.Spec.ClusterIP,
 	}
-	ipList = append(ipList, tc.Spec.TiDB.ExtraSANIPList...)
+	if tc.Spec.TiDB.TLSClient.AutoGenerated != nil {
```

Comment:
Could you merge this `if` condition with the previous, when the cert is auto generated, both the `ipList` and `hostList` need to be appended.